### PR TITLE
s4640661: Semi-supervised node classification of the Facebook large page-page network dataset

### PR DESCRIPTION
**Pull request by kristiannorved**
We use a multi-layer graph-convolutional network to classify the Facebook large page-page network dataset with 95.25% testing accuracy. Essentially, the given dataset consists of nodes (vertices) and edges between nodes. Each node represents an official Facebook page, and edges represent mutual likes between  two pages. Each node has 128 features associated with it, and belongs to one of four classes: politicians, government organizations, television shows, and companies. The completed task was to be able to predict with high accuracy which of these four classes any given node in the dataset belongs to.

New files:
* `dataset.py` loads the processes the raw dataset
* `modules.py` stores the graph convolutional neural network as a class
* `train.py` when run will perform 100 epochs of training on the given dataset
* `predict.py` when run will perform testing classification as well as visualise the results of the model
* `run.py` is a script to run the whole program

### CODE REVIEW

**Comment by gayanku** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Normal

Readme Overall: Good

- Project Overview: Good
- Model: Good
- Data + Preprocessing: Ok, include info about data - edges, feature dims etc.
- Training / Loss Curve(s): Good
- Result Demonstration: Good
- References: Good

Functionality:

- Multi-layer GCN: OK. **uses external library **  (torch_geometric.nn import GCNConv). Could use ealy stop.
- Semi supervised: Good
- Accuracy: Good
- TSNE or UMAP: **Fair.** Could use a different color rather than white for points, as the background is also white.
- Discussion: **OK**. Given that there are 4 classes, stating the presence of outliers (ex " the class clustering heavily by eliminating the vast majority of class outliers.") is incorrect.

Code:

- Consistent with Results: Yes
- File Structure: Good
- Commenting: Good
- Commit frequency: Good
- Commit messages: Good

Other Comments:
N/A

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Normal -5
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable, missing usage -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
